Add Methylation tab to Comparison page and Comparison tab

dippindots opened the pull request:
Fix cBioPortal/cbioportal#7683
Add Methylation tab to Comparison page and Comparison tab
Describe changes proposed in this pull request:
- Comparison tab
![image](https://user-images.githubusercontent.com/15748980/87701976-c42b8700-c766-11ea-9ff4-09fa78d4b508.png)
- Comparison page
![image](https://user-images.githubusercontent.com/15748980/87702048-e6bda000-c766-11ea-929a-732e92ed3907.png)

dippindots commented:
@kalletlak @adamabeshouse Can I get your review on this pr? This pr is adding another tab `Methylation` on result view comparison tab and comparison page.

adamabeshouse commented on src/shared/lib/comparison/ComparisonStore.ts:
```diff
@@ -943,7 +991,7 @@ export default class ComparisonStore {
         referenceGenesPromise: this.hugoGeneSymbolToReferenceGene,
         getSelectedProfileMap: () =>
-            this.selectedProteinEnrichmentProfileMap.result!, // returns an empty array if the selected study doesn't have any mRNA profiles
+            this.selectedProteinEnrichmentProfileMap.result!, // returns an empty array if the selected study doesn't have any protein profiles
```
nice thanks for cleaning that up

kalletlak commented on src/pages/resultsView/enrichments/EnrichmentsUtil.tsx:
```diff
@@ -423,6 +423,17 @@ export function pickProteinEnrichmentProfiles(profiles: MolecularProfile[]) {
+export function pickMethylationEnrichmentProfiles(
+    profiles: MolecularProfile[]
+) {
+    const methylationProfiles = profiles.filter(p => {
```
directly return instead of creating local variable

dippindots commented:
@cBioPortal/product Can I get your review of this pr? It's adding the Methylation tab to the Comparison page and Comparison tab. Here are two example links: [Comparison tab](https://deploy-preview-3312--cbioportalfrontend.netlify.app/results/comparison?Action=Submit&RPPA_SCORE_THRESHOLD=2.0&Z_SCORE_THRESHOLD=2.0&cancer_study_list=brca_tcga&case_set_id=brca_tcga_cnaseq&data_priority=0&gene_list=TP53&geneset_list=%20&genetic_profile_ids_PROFILE_COPY_NUMBER_ALTERATION=brca_tcga_gistic&genetic_profile_ids_PROFILE_MUTATION_EXTENDED=brca_tcga_mutations&profileFilter=0&tab_index=tab_visualize&comparison_subtab=methylation
) and [Comparison page](https://deploy-preview-3312--cbioportalfrontend.netlify.app/comparison/methylation?sessionId=5ce411c7e4b0ab4137874076)

dippindots commented:
@schultzn Thank you for reviewing this pr, I will make change to the first and second points.
- [ ] table adding a third digit (we are showing two currently)
- [ ] change the label for the last column “Higher methylation…”

I am not sure about the third one, @adamabeshouse not sure do you know the answer for this question? `Lastly, did we ever have the functionality to summarize the different values for the box and whiskers (mean, median, quartiles, etc.) upon mouseover?`

adamabeshouse commented:
@dippindots I'm not sure if we ever had it but I'm pretty sure that data already exists so it should be relatively simple to put it into the tooltip

dippindots commented:
@adamabeshouse Thanks for confirming this!
@schultzn Although we don't have the `functionality to summarize the different values for the box and whiskers (mean, median, quartiles, etc.) upon mouseover` before, but we can add this for sure.

@cBioPortal/product I have two questions regarding to the requested changes:
1. Which column do we want to keep the third digit? And should I apply that to `mrna` and `protein` tab?
[Comparison tab methylation example](https://deploy-preview-3312--cbioportalfrontend.netlify.app/results/comparison?Action=Submit&RPPA_SCORE_THRESHOLD=2.0&Z_SCORE_THRESHOLD=2.0&cancer_study_list=brca_tcga&case_set_id=brca_tcga_cnaseq&data_priority=0&gene_list=TP53&geneset_list=%20&genetic_profile_ids_PROFILE_COPY_NUMBER_ALTERATION=brca_tcga_gistic&genetic_profile_ids_PROFILE_MUTATION_EXTENDED=brca_tcga_mutations&profileFilter=0&tab_index=tab_visualize&comparison_subtab=methylation) [Comparison page methylation example](https://deploy-preview-3312--cbioportalfrontend.netlify.app/comparison/methylation?sessionId=5ce411c7e4b0ab4137874076) [Comparison tab mrna example](https://deploy-preview-3312--cbioportalfrontend.netlify.app/results/comparison?Action=Submit&RPPA_SCORE_THRESHOLD=2.0&Z_SCORE_THRESHOLD=2.0&cancer_study_list=brca_tcga&case_set_id=brca_tcga_cnaseq&comparison_subtab=mrna&data_priority=0&gene_list=TP53&geneset_list=%20&genetic_profile_ids_PROFILE_COPY_NUMBER_ALTERATION=brca_tcga_gistic&genetic_profile_ids_PROFILE_MUTATION_EXTENDED=brca_tcga_mutations&profileFilter=0&tab_index=tab_visualize) [Comparison page mrna example](https://deploy-preview-3312--cbioportalfrontend.netlify.app/comparison/mrna?sessionId=5ce411c7e4b0ab4137874076)

2. Any suggestions to add a tooltip for `summarize the different values for the box and whiskers (mean, median, quartiles, etc.)` upon mouseover.

I was wondering if you might be able to give me some advice.

kalletlak commented:
@dippindots as @adamabeshouse suggested data should be already present, just tooltip needs to be updated. We already use it in box plots of clinical tab in comparison page. Ex click row with `Kruskal Wallis Test` in https://www.cbioportal.org/comparison/clinical?sessionId=5cf89323e4b0ab413787436c

schultzn commented:
That’s what I remember! Thanks!

It would be nice if we could support this throughout cBioPortal.


> On Jul 22, 2020, at 11:35 AM, Karthik Kalletla <notifications@github.com> wrote:
> 
> 
> @dippindots <https://github.com/dippindots> as @adamabeshouse <https://github.com/adamabeshouse> suggested data should be already present, just tooltip needs to be updated. We already use it in box plots of clinical tab in comparison page. Ex click row with Kruskal Wallis Test in https://www.cbioportal.org/comparison/clinical?sessionId=5cf89323e4b0ab413787436c <https://www.cbioportal.org/comparison/clinical?sessionId=5cf89323e4b0ab413787436c>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/cBioPortal/cbioportal-frontend/pull/3312#issuecomment-662524294>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AC2XPYETFPDNQ5LVM54NFEDR44BLXANCNFSM4O4TXOVA>.
>

tmazor commented:
@dippindots what an awesome addition!

A few things I noticed:
1) The tooltip on the gene box plot says "mRNA expression" but should say "DNA methylation"
![image](https://user-images.githubusercontent.com/10508276/88291869-dd808600-ccc6-11ea-900c-c56d00fc7153.png)
2) My preference is to call the tab "DNA Methylation" rather than just "Methylation" (because in my experience people often think of histone methylation first) - but I'm also ok to leave it as is if that's the general preference. @schultzn @jjgao ?
3) The tooltips of the mu & sigma columns still refer to expression
![image](https://user-images.githubusercontent.com/10508276/88292390-98108880-ccc7-11ea-9d27-7a3e56aaefab.png)
![image](https://user-images.githubusercontent.com/10508276/88292421-9c3ca600-ccc7-11ea-8406-a453de2f775c.png)
4) If I download or copy the table data, the mu and sigma columns are just called by the group name, which means a user ends up with columns with identical names that have different meanings
![image](https://user-images.githubusercontent.com/10508276/88292814-1ec56580-ccc8-11ea-9d92-b6aeb67d628b.png)
Can we update the column names to include mu/sigma or the words "Mean in XXX" "Standard Deviation in XXX"
Also this might be true of other tabs (I didn't test) - if so, can we fix all of them?

dippindots commented:
@tmazor @schultzn Thanks for your time! I have addressed all comments related to the `DNA Methylation` tab and created a new issue (https://github.com/cBioPortal/cbioportal/issues/7737) to log the improvement we want to make for all three enrichment tabs, I will address those comments in another pr. Can I get your review again on this pr?

tmazor commented:
Looking good @dippindots ! Just a few more tweaks to the tooltips:

1) This data is not log2, so we need to update all the mean/sd tooltips to just say "Mean methylation" and "Standard deviation of methylation"
![image](https://user-images.githubusercontent.com/10508276/88719390-37cb7d80-d0f1-11ea-9842-1c70c4f512d6.png)
![image](https://user-images.githubusercontent.com/10508276/88719404-3e59f500-d0f1-11ea-8790-7c747a098025.png)
2) Similarly the Log Ratio tooltip should remove "(unlogged)"
![image](https://user-images.githubusercontent.com/10508276/88719444-4ca81100-d0f1-11ea-89e4-1b0c002f7a36.png)
3) This might belong in the follow-up issue, but for mRNA & Protein & this tab, can we capitalize Group in the lower label on the x-axis of the plot on the right:
![image](https://user-images.githubusercontent.com/10508276/88719367-2bdfbb80-d0f1-11ea-8301-211f7dd73419.png)

dippindots commented:
Hi @tmazor, all the comments addressed. Thanks you for taking time to help me identify the potential misleading (incorrect) information. Again, thanks for your time and can get your review again?

tmazor commented:
Looks good to me @dippindots !

My only lingering question is about the data itself... I assume this is raw beta values directly from the array? @jjgao do you know if there's been any processing (besides selecting a single probe per gene)? In my experience this data should really be normalized & transformed before doing these kinds of analyses.

jjgao commented:
@schultzn I've create another issue to capture whisker values discussion: https://github.com/cBioPortal/cbioportal/issues/7757

@tmazor I think they are just raw beta values. We pick the probe that was most negatively correlated with the expression. Would you please elaborate on normalization and transformation - maybe create another issue on datahub? It's good timing to do that b/c we are curating the pancancer atlas dataset.

@dippindots let's merge this. Thanks for the good work!

tmazor commented:
@jjgao I made a ticket on the datahub board: https://github.com/cBioPortal/datahub/issues/1210

alisman commented on src/pages/resultsView/enrichments/EnrichmentsUtil.tsx:
```diff
@@ -535,20 +545,24 @@ export function getAlterationEnrichmentColumns(
     groups: { name: string; description: string; color?: string }[],
-    alteredVsUnalteredMode?: boolean
+    alteredVsUnalteredMode?: boolean,
+    isMethylation?: boolean
```
could this instead be the enum type?  That's more general/flexible

alisman commented on src/pages/resultsView/enrichments/ExpressionEnrichmentsContainer.tsx:
```diff
@@ -210,7 +213,11 @@ export default class ExpressionEnrichmentContainer extends React.Component<
             columns.push(ExpressionEnrichmentTableColumnType.TENDENCY);
         } else {
-            columns.push(ExpressionEnrichmentTableColumnType.EXPRESSED);
+            if (this.props.isMethylation) {
```
again, if we can generalize this rather than have isMethylation, it would be better

alisman commented on src/shared/lib/comparison/ComparisonStore.ts:
```diff
@@ -534,6 +554,27 @@ export default class ComparisonStore {
+    readonly selectedMethylationEnrichmentProfileMap = remoteData({
+        await: () => [this.methylationEnrichmentProfiles],
+        invoke: () => {
+            if (_.isEmpty(this._methylationEnrichmentProfileMap)) {
```
put in a comment here explaining this if statement
